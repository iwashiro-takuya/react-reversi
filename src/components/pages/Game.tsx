import React, { useState } from 'react';
import styled from 'styled-components';
import Board from '../organisms/Board';
import Const from '../../const';

const { Color } = Const;

const Wrapper = styled.div`
  width: 1024px;
  margin: 0 auto;
  color: ${Color.TX_BLACK};
`;

const Game: React.FC = () => {
  const [isGameStart, setGameStart] = useState(true); // TODO:最後にfalseにする

  return (
    <Wrapper>
      {!isGameStart && (
        <button
          type="button"
          onClick={() => {
            setGameStart(true);
          }}
        >
          Game Start
        </button>
      )}
      {isGameStart && <Board onSideSquares={8} />}
    </Wrapper>
  );
};

export default Game;
