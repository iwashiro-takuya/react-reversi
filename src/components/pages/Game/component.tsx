import React from 'react';
import styled from 'styled-components';
import Button from 'src/components/atoms/Button/component';
import Board from 'src/components/organisms/Board/component';
import Information from 'src/components/organisms/Information/component';
import Const from 'src/const';
import useGame from './hooks';

const { Color } = Const;

export interface GameProp {
  dataCy: string;
}
const Game: React.FC<GameProp> = ({ dataCy }) => {
  const { isGameStart, startGame } = useGame();

  return (
    <StyledGameWrapper data-cy={dataCy}>
      {isGameStart ? (
        <>
          <Board dataCy="board" />
          <Information />
        </>
      ) : (
        <Button
          onClick={() => {
            startGame(8);
          }}
          text="Game Start"
          dataCy="start"
        />
      )}
    </StyledGameWrapper>
  );
};

const StyledGameWrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  width: 992px;
  height: 768px;
  margin: 0 auto;
  color: ${Color.TX_BLACK};
`;

export default Game;
