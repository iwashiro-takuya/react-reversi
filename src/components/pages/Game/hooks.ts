import { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { setGameStartFlg, setBoardStates } from 'src/redux/modules/game';

const useGame = (): {
  isGameStart: boolean;
  onGameStart: (sideSquaresCount: number) => void;
} => {
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(setGameStartFlg(false));
  }, [dispatch]);

  const onGameStart = (sideSquaresCount: number) => {
    dispatch(setGameStartFlg(true));
    dispatch(setBoardStates(sideSquaresCount));
  };

  const isGameStart = useSelector(
    (state: StoreState) => state.game.isGameStart,
  );

  return {
    isGameStart,
    onGameStart,
  };
};
export default useGame;
