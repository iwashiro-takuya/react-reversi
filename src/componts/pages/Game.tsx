import React from 'react';
import styled from 'styled-components';
import Const from '../../const';

import Board from '../organisms/Board';

const { Color } = Const;

const Wrapper = styled.div`
  color: ${Color.TX_BLACK};
  margin: 0 auto;
  width: 1024px;
`;

const Game: React.FC = () => (
  <Wrapper>
    <Board squares={64} />
  </Wrapper>
);

export default Game;
