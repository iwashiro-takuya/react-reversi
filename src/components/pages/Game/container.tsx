import React, { useState } from 'react';
import Game, { GameProp } from 'src/components/pages/Game/component';
import Button from 'src/components/atoms/Button/component';
import Board from 'src/components/organisms/Board/container';
import Information from 'src/components/organisms/Information/component';

const EnhancedGame: React.FC<GameProp> = ({ dataCy }) => {
  const [isGameStartFlg, setGameStart] = useState(false);

  return (
    <Game dataCy={dataCy}>
      {isGameStartFlg ? (
        <>
          <Board onSideSquares={8} dataCy="board" />
          <Information />
        </>
      ) : (
        <Button
          onClick={() => {
            setGameStart(true);
          }}
          text="Game Start"
          dataCy="start"
        />
      )}
    </Game>
  );
};

export default EnhancedGame;
