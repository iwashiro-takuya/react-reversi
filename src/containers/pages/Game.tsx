import React, { useState } from 'react';
import Game from 'src/components/pages/Game';
import Button from 'src/containers/atoms/Button';
import Board from 'src/containers/organisms/Board';

const EnhancedGame: React.FC = () => {
  const [isGameStartFlg, setGameStart] = useState(false);

  return (
    <Game>
      {isGameStartFlg ? (
        <Board onSideSquares={8} />
      ) : (
        <Button
          onClick={() => {
            setGameStart(true);
          }}
          text="Game Start"
        />
      )}
    </Game>
  );
};

export default EnhancedGame;
