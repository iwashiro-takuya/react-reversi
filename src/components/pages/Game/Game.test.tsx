import React from 'react';
import renderer from 'react-test-renderer';
import Game from './container';

it('Game snapshot testing', () => {
  const component = renderer.create(<Game dataCy="start" />);
  expect(component).toMatchSnapshot();
});
