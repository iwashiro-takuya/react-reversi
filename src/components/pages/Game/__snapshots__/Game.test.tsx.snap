// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game unit test snapshot 1`] = `
<div
  className="sc-bdnylx ljgJuJ"
  data-cy="start"
>
  <button
    className="sc-gtssRu uftyw"
    data-cy="start"
    onClick={[Function]}
    type="button"
  >
    Game Start
  </button>
</div>
`;
